lua/compe/init.lua

```diff
@@ -1,84 +1,128 @@
+--- registers a new vim source
+-- passes params using new func from `source` and `vim_bridge` to `completion`
+-- register_source func.
+-- @param id string representing the source name.
+-- @function register_vim_source
+compe.register_vim_source = function(self, id)
+  self.completion:register_source(source.new(id, vim_bridge.new(id)))
 end
 
---- unregister_source
-function Compe.unregister_source(self, id)
+--- unregisters a vim or lua source
+-- passes params using to `completion` unregister_source func.
+-- @param id string representing the source name.
+-- @function unregister_source
+compe.unregister_source = function(self, id)
   self.completion:unregister_source(id)
 end
 
---- on_complete_changed
-function Compe.on_complete_changed(self)
+--- wraps `completion:on_complete_changed`
+-- ???
+-- @param self
+-- @function on_complete_changed
+compe.on_complete_changed = function(self)
   local status, value = pcall(function() self.completion:on_complete_changed() end)
   if not(status) then
-    Debug:log(value)
+    debug:log(value)
   end
 end
 
---- on_complete_done
-function Compe.on_complete_done(self)
+--- wraps `completion:on_complete_done`
+-- ???
+-- @param self
+-- @function on_complete_done
+compe.on_complete_done = function(self)
   local status, value = pcall(function() self.completion:on_complete_done() end)
   if not(status) then
-    Debug:log(value)
+    debug:log(value)
   end
 end
 
---- on_text_changed
-function Compe.on_text_changed(self)
+--- wraps `completion:on_text_changed`
+-- Event that triggers on text change ???
+-- @param self
+-- @function on_text_changed
+compe.on_text_changed = function(self)
   local status, value = pcall(function() self.completion:on_text_changed() end)
   if not(status) then
-    Debug:log(value)
+    debug:log(value)
   end
 end
 
---- on_insert_leave
-function Compe.on_insert_leave(self)
+--- wraps `completion:on_text_changed`
+-- Event that triggers on insert leave ???
+-- @param self
+-- @function on_insert_leave
+compe.on_insert_leave = function(self)
   local status, value = pcall(function() self.completion:on_insert_leave() end)
   if not(status) then
-    Debug:log(value)
+    debug:log(value)
   end
 end
 
---- on_manual_complete
-function Compe.on_manual_complete(self)
+--- wraps `completion:on_manual_complete`
+-- ???
+-- @param self
+-- @function on_manual_complete
+compe.on_manual_complete = function(self)
   local status, value = pcall(function() self.completion:on_manual_complete() end)
   if not(status) then
-    Debug:log(value)
+    debug:log(value)
   end
 end
 
--- add_history
-function Compe.add_history(self, word)
+--- wraps `completion:add_history`
+-- ???
+-- @param self
+-- @function add_history
+compe.add_history = function(self, word)
   self.completion:add_history(word)
 end
 
---- clear
-function Compe.clear(self)
+--- wraps `completion:clear`
+-- ???
+-- @param self
+-- @function clear
+compe.clear = function(self)
```

Comment:
Clear all current completion status.
It is useful to realize vim's `<C-e>` mapping.